crates/red_knot/src/program/check.rs

```diff
@@ -39,7 +39,30 @@ impl Program {
     ) -> Result<Diagnostics, CheckError> {
         context.cancelled_ok()?;
 
-        // TODO schedule the dependencies.
+        let symbol_table = self.symbol_table(file);
+        let dependencies = symbol_table.dependencies();
+
+        if !dependencies.is_empty() {
+            let module_name = self.file_to_module(file).map(|module| module.name(self));
+
+            // TODO scheduling all dependencies here is wasteful if we don't infer any types on them
+            //  but I think that's unlikely, so it is okay?
+            //  Anyway, we need to figure out a way to retrieve the dependencies of a module
+            //  from the persistent cache. So maybe it should be a separate query after all.
+            for dependency in dependencies {
+                let dependency_name = dependency.module_name(module_name.as_ref());
+
+                if let Some(dependency_name) = dependency_name {
+                    // TODO The resolver doesn't support native dependencies yet.
+                    if let Some(dependency) = self.resolve_module(dependency_name) {
+                        if dependency.path(self).root().kind().is_first_party() {
+                            context.schedule_check_file(dependency.path(self).file());
```

Comment:
We may want to distinguish between `schedule_check_file` and `schedule_index_file`. For non-first-party dependencies, there's no point in scheduling a check of them (since we won't check them) but we may still want to eagerly schedule indexing them (that is, parsing and creating symbol tables).

Reply:
Yeah, that might be good. The scheduling is still very much unclear to me. So I'll defer figuring out the right solution for now.

Reply:
Maybe a TODO here then?

Reply:
I rephrased the native dependencies todo